lib/Dialect/TT/IR/TTOpsTypes.cpp

```diff
@@ -394,10 +394,12 @@ static mlir::AffineMap collapsedLinearAffineMap(
     if (end < 0) {
       end += shape.size();
     }
+    if (end < 0) {
+      continue;
+    }
```

Comment:
I'm curious when does this happen? It seems to me that valid value for begin/end must be in interval (-shape_size, shape_size).